donjo-app/controllers/Keluar.php

```diff
@@ -173,21 +170,13 @@ public function graph()
 		$this->render('surat/surat_keluar_graph', $data);
 	}
 
-	public function filter()
-	{
-		$filter = $this->input->post('filter');
-		if ($filter != 0)
-			$_SESSION['filter'] = $filter;
-		else unset($_SESSION['filter']);
-		redirect('keluar');
-	}
-
-	public function jenis()
+	public function filter($filter)
 	{
-		$jenis = $this->input->post('jenis');
-		if (!empty($jenis))
-			$_SESSION['jenis'] = $jenis;
-		else unset($_SESSION['jenis']);
+		$value = $this->input->post($filter);
+		if ($value == 'tahun') $this->session->unset_userdata('bulan'); //hapus filter bulan
```

Comment:
@apidong terima kasih. Apakah anda lakukan testing? Seharusnya:

`if ($filter == 'tahun') $this->session->unset_userdata('bulan');`

Reply:
Oh ya lupa pak . 😁